cpp/src/stream_compaction/filter/filter.cu

```diff
@@ -0,0 +1,351 @@
+                              num_selected,
+                              stencil_iterator,
+                              stream,
+                              mr);
+    return cudf::make_strings_column(filtered, cudf::string_view{nullptr, 0}, stream, mr);
+  }
+
+  template <typename T>
+  std::unique_ptr<cudf::column> operator()(cudf::column_device_view const& col,
+                                           cudf::size_type num_selected,
+                                           cudf::size_type const* stencil_iterator,
+                                           rmm::cuda_stream_view stream,
+                                           rmm::device_async_resource_ref mr) const
+  {
+    CUDF_FAIL("Unsupported column type for filter operation: " +
+                cudf::type_to_name(cudf::data_type{cudf::type_to_id<T>()}),
+              std::invalid_argument);
+  }
+};
+
+std::unique_ptr<cudf::column> filter_column(
+  cudf::column_device_view const& column,
+  cudf::size_type num_selected,
+  cudf::jit::device_span<cudf::size_type const> filter_indices,
+  rmm::cuda_stream_view stream,
+  rmm::device_async_resource_ref mr)
+{
+  return cudf::type_dispatcher<dispatch_storage_type>(column.type(),
+                                                      filter_dispatcher<false>{},
+                                                      column,
+                                                      num_selected,
+                                                      filter_indices.begin(),
+                                                      stream,
+                                                      mr);
+}
+
+void launch_filter_kernel(jitify2::ConfiguredKernel& kernel,
+                          cudf::jit::device_span<cudf::size_type> output,
+                          std::vector<column_view> const& input_columns,
+                          std::optional<void*> user_data,
+                          rmm::cuda_stream_view stream,
+                          rmm::device_async_resource_ref mr)
+{
+  auto outputs = cudf::jit::to_device_vector(
+    std::vector{cudf::jit::device_optional_span<cudf::size_type>{output, nullptr}}, stream, mr);
+
+  auto [input_handles, inputs] =
+    cudf::jit::column_views_to_device<column_device_view, column_view>(input_columns, stream, mr);
+
+  cudf::jit::device_optional_span<cudf::size_type> const* outputs_ptr = outputs.data();
+  column_device_view const* inputs_ptr                                = inputs.data();
+  void* p_user_data                                                   = user_data.value_or(nullptr);
+
+  std::array<void*, 3> args{&outputs_ptr, &inputs_ptr, &p_user_data};
+
+  kernel->launch(args.data());
+}
+
+void perform_checks(column_view base_column,
+                    std::vector<column_view> const& columns,
+                    std::optional<std::vector<bool>> copy_mask)
+{
+  CUDF_EXPECTS(is_runtime_jit_supported(), "Runtime JIT is only supported on CUDA Runtime 11.5+");
```

Comment:
We no longer need to perform checks for 11.5+ because we have a lower bound of 12.0.

Reply:
(This can be deferred to a follow-up PR if you wish.)